### ISSUE TITLE
6.0: Unknown variable: 'sess.timeout_idle' (Only available when VCL syntax)

### ISSUE THREAD

**Issue by Dridi** [varnishcache/varnish-cache#3564]
This does not happen in trunk, but it happens in the 6.0 LTS branch currently.

It's not possible to use `sess.timeout_idle` with the 4.0 syntax, despite its lack of such restriction. I suspect that the symbol `sess.xid` declared just before and restricted to 4.1 syntax might "leak" this requirement via the intermediate `sess` symbol.

I haven't researched this, so this is purely speculative.

**Comment by Dridi** [comment]
The same problem existed with #2981 when it was merged, is has been fixed later (searching).

**Comment by Dridi** [comment]
It was accidentally fixed by bc3b93bcca6f5a573eb1344d8c374124b0e8aecb.

Verified with this test case:

```
varnishtest "Check timeout_idle"

varnish v1 -arg "-p timeout_idle=1" -syntax 4.0 -vcl {
        backend dummy { .host = "${bad_ip}"; }

        sub vcl_deliver {
                if (req.url == "/sess") {
                        set sess.timeout_idle = 2s;
                }
        }
        sub vcl_backend_error {
                set beresp.status = 200;
                set beresp.ttl = 1h;
        }
} -start

client c1 {
        txreq
        rxresp
        delay 0.2
        txreq
        rxresp
        delay 1.2
        expect_close
} -run

client c1 {
        txreq -url "/sess"
        rxresp
        delay 1.2
        txreq
        rxresp
        delay 2.2
        expect_close
} -run
```

This is basically b67 with a 4.0 syntax.

**Comment by Dridi** [comment]
I can confirm that moving `sess.xid` down is enough pass the test case above:

```diff
--- doc/sphinx/reference/vcl_var.rst
+++ doc/sphinx/reference/vcl_var.rst
@@ -1172,14 +1172,6 @@ transactions may take place. It may comprise the traffic over an
 HTTP/1 keep-alive connection, or the multiplexed traffic over an
 HTTP/2 connection.
 
-sess.xid       ``VCL >= 4.1``
-
-       Type: STRING
-
-       Readable from: client, backend
-
-       Unique ID of this session.
-
 sess.timeout_idle
 
        Type: DURATION
@@ -1191,6 +1183,14 @@ sess.timeout_idle
        Idle timeout for this session, defaults to the
        ``timeout_idle`` parameter, see :ref:`varnishd(1)`
 
+sess.xid       ``VCL >= 4.1``
+
+       Type: STRING
+
+       Readable from: client, backend
+
+       Unique ID of this session.
+
 storage
 ~~~~~~~
 
```

Not sure about potential regressions related to that.

**Comment by Dridi** [comment]
Fixed by #3656.
